README.md

```diff
@@ -196,11 +196,65 @@ Configurable middleware arguments include:
   The argument was added for cases where users might want to alter incoming or generated ID values in some way. It
   provides a mechanism for transforming an incoming ID in a way you see fit. See the middleware code for more context.
 
+## CORS
+
+If you are using cross-origin resource sharing ([CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)), e.g.
+you are making requests to an API from a frontend JavaScript code served from a different origin, you have to ensure
+two things:
+
+- permit correlation ID header in the incoming requests' HTTP headers so the value can be reused by the middleware,
+- add the correlation ID header to the allowlist in responses' HTTP headers so it can be accessed by the browser.
+
+This can be best accomplished by using a dedicated middleware for your framework of choice. Here are some examples.
+
+### Starlette
+
+Docs: https://www.starlette.io/middleware/#corsmiddleware
+
+```python
+from starlette.applications import Starlette
+from starlette.middleware import Middleware
+from starlette.middleware.cors import CORSMiddleware
+
+
+middleware = [
+    Middleware(
+        CORSMiddleware,
+        allow_origins=['*'],
+        allow_methods=['*'],
+        allow_headers=['X-Request-ID'],
+        expose_headers=['X-Requested-With', 'X-Request-ID']
```

Comment:
Why `X-Requested-With`?

Reply:
It's a header set by most AJAX libraries which send requests with `XMLHttpRequest` method. Think jQuery ;-)

Reply:
Anyway, good catch! It should be included in `allow_headers` (as it's a request header), not `expose_headers`. Will correct it right away.

Reply:
@JonasKs fixed.